07-routing-gossip.md

```diff
@@ -312,6 +324,19 @@ The origin node:
   - MUST set `features` according to [BOLT #9](09-features.md#assigned-features-flags)
   - SHOULD set `flen` to the minimum length required to hold the `features`
   bits it sets.
+ - if supports `option_will_fund`:
+  - MUST include `option_will_fund` tlv type
+  - MUST set `funding_fee_base_sat` to the base fee (in satoshi) it will charge
+    for any reciprocated funding.
+  - MUST set `funding_fee_proportional_basis` to the amount (in
+    thousandths of a satoshi) it will charge per contributed satoshi.
```

Comment:
Why not millionths?

Reply:
Because this fits better in u16?  As these msgs are gossiped all over the network, bytes matter.  Though basis points would be 10,000th, so perhaps rename to "funding_fee_proportional_thousanths" explicitly?  (Allowing up to 65% seems nice; 6.5% is probably a bit low).

Reply:
... names don't match fields, particularly missing _max?

Reply:
Good point, my thinking was to be consistent and sufficiently granular. I would like 10000ths more than 1000ths.

Reply:
I updated this to say 'thousandths' -- generally it's advisable to expose these to end users as ppm though -- using parts-per-thousandths is a shortcut to reduce the size of the node_announcement TLV

Reply:
@niftynei do you mean that when exposed to user as ppm some rounding will be implemented or a more precise value will be negotiated when actually opening the channel?

Reply:
The channel fee is expressed as "channel_fee_proportional_millionths" and is broadcast via the "channel_update" gossip message. The channel lease commits to a "channel_fee_max_proportional_thousandths". Let's call these "ppm" and "ppt" for short. 1 ppt is equal to 1,000ppm. Practically, setting a 'channel_fee_max_proportional_thousandth" to 1 would be a commitment to never set the "channel_fee_max_proportional_millionth" larger than 1000.